csrc/scheduler/matmul_hopper+.cpp

```diff
@@ -566,17 +584,81 @@ void HopperPlus::scheduleMmaResults() {
     }
 
     transformLikeMmaOutputWithK(mma_result);
+    setMmaResultAllocationDomain(mma_result);
 
-    auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
-        mma_result->getLoopDomain());
-    mma_result->setAllocationDomain(s.as<IterDomain*>(), true);
     mma_result->axis(-1)->parallelize(ParallelType::Mma);
     mma_result->axis(-2)->parallelize(ParallelType::Mma);
     mma_result->axis(-3)->parallelize(ParallelType::Mma);
   }
 }
 
-void HopperPlus::scheduleEpilogueWithoutSmemEpilogue() {
+void HopperPlus::scheduleEpilogueWithoutSmemEpilogueBlackwell() {
+  std::vector<TensorView*> cached_tvs;
+  std::vector<TensorView*> propagate_to =
+      splitk_sums_.empty() ? mma_results_ : splitk_sums_;
+  std::vector<TensorView*> tmem_ld_tvs;
+  for (auto mma_result : mma_results_) {
+    TensorView* tmem_ld_tv = cacheAfter(mma_result);
+    tmem_ld_tv->definition()->as<LoadStoreOp>()->setOpType(
+        LoadStoreOpType::LdTMem);
+    tmem_ld_tvs.push_back(tmem_ld_tv);
+  }
+  for (auto& [c, c_cache] : cached_epilogue_inputs_) {
+    cached_tvs.push_back(c_cache);
+    propagate_to.push_back(c);
+  }
+  for (Val* dv : fusion_->outputs()) {
+    TensorView* d = dv->as<TensorView>();
+    NVF_ERROR(d->definition() && d->definition()->isA<LoadStoreOp>());
+
+    // Apply the default scheduling that is common to all register
+    // TensorViews after wgmma.
+    blockTileTensors({d});
+    parallelizeBlocks({d});
+    transformLikeMmaOutputWithoutK(d);
+
+    d->axis(-2)->parallelize(ParallelType::TIDx);
+    int64_t tmem_vectorize_factor = 1;
+    const int64_t n_mma = getN(params_->mma_macro);
+    while (n_mma % tmem_vectorize_factor == 0 && tmem_vectorize_factor <= 128) {
+      tmem_vectorize_factor *= 2;
+    }
+    if (tmem_vectorize_factor < n_mma) {
+      d->split(-1, tmem_vectorize_factor);
+    }
+
+    // TODO: We need to check bank conflicts in this path.
+    // Propagate schedule changes back to the outputs of the Mma op.
+    scheduler_utils::BoundedDirectionalTransformPropagator::backward(
+        d,
+        -1,
+        propagate_to,
+        scheduler_utils::BoundedDirectionalTransformPropagator::Options()
+            .propagateParallelType());
+
+    // TODO: Support vectorization_factor in MatmulParams
+    if (tmem_vectorize_factor > params_->supported_vec_size.epilogue) {
+      d->split(-1, params_->supported_vec_size.epilogue);
+      for (auto c : cached_tvs) {
+        bool is_2d_epilogue_input =
+            TensorDomain::noBroadcasts(c->domain()->logical()).size() == 2;
+        if (is_2d_epilogue_input) {
+          c->split(-1, params_->supported_vec_size.epilogue);
+        }
+      }
+    }
+    d->axis(-1)->parallelize(ParallelType::Vectorize);
```

Comment:
Documentation idea:
`// After epilogue_vec_size: [..., Mo * No, Mw, Nw, Mi, Ni / tmem_vectorize_factor, tmem_vectorize_factor / epilogue_vec_size, epilogue_vec_size]`

Reply:
Added a comment in:

```C++
    // Vectorize the epilogue input load and output store. TMem load can
    // be vectorized to 512 byte, but gmem load/store can only be vectorized
    // to 16 bytes. So we need to further split the last dimension and use
    // multiple vector loads/stores. for each TMem load/store.
```